src/rngs/os.rs

```diff
@@ -188,10 +188,31 @@ impl RngCore for OsRng {
 
     fn try_fill_bytes(&mut self, dest: &mut [u8]) -> Result<(), Error> {
         if dest.len() == 0 { return Ok(()); }
-        self.0.try_fill_bytes(dest)
+        let max = self.0.max_chunk_size().unwrap_or(dest.len());
+        if dest.len() <= max {
+            trace!("OsRng: reading {} bytes via {}",
+                   dest.len(), self.0.method_str());
+        } else {
+            trace!("OsRng: reading {} bytes via {} in {} chunks of {} bytes",
+                   dest.len(), self.0.method_str(), (dest.len() + max) / max, max);
+        }
+        for slice in dest.chunks_mut(max) {
+            self.0.fill_chunk(slice)?;
+        }
+        Ok(())
     }
 }
 
+trait OsRngImpl where Self: Sized {
+    fn new() -> Result<Self, Error>;
+    fn fill_chunk(&mut self, dest: &mut [u8]) -> Result<(), Error>;
+    fn max_chunk_size(&self) -> Option<usize> { None }
```

Comment:
May as well just use `core::usize::MAX` and drop the `Option` since this is the default impl.